src/pages/trainer/training/edit/components/editor.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import {ToolbarContainerComponent} from './toolbar';
+const classNames: any = require('./editorStyles.scss');
+
+interface IProps {
+  content: string;
+  cursorStartPosition: number;
+  shouldSetEditorFocus: boolean;
+  className: string;
+  onContentChange: (content: string) => void;
+  initializeEditor: (editor: HTMLTextAreaElement) => void;
+  updateEditorCursor: (editor: HTMLTextAreaElement, cursorStartPosition: number) => void;
+}
+
+export class EditorComponent extends React.Component<IProps, {}> {
+  private editor: HTMLTextAreaElement;
+  private refHandlers = {
+    textArea: (textArea) => { this.editor = textArea; },
+  };
+
+  public componentDidMount() {
+    this.props.initializeEditor(this.editor);
+  }
+
+  public componentDidUpdate() {
+    if (this.props.shouldSetEditorFocus) {
```

Comment:
shouldSetEditorFocus is a one time prop, where do we reset this property once the focus thing has been updated?

I remember discussing this with @pablonete we came with two possible approaches:

- DesiredFocus 
- Using a unique number to force  trigger.

As far as I see here componentDidUpdate executes on every cycle we need more dicusssion on this.